proteus/mprans/GN_SW2DCV.h

```diff
@@ -1842,24 +1420,31 @@ class GN_SW2DCV : public GN_SW2DCV_base {
           alpha_numerator += hj - hi;
           alpha_denominator += fabs(hj - hi);
 
+          // define dij_small in j loop
+          double x = fabs(Cx[ij]) + fabs(Cy[ij]);
+          dij_small = fmax(dij_small, x * speed);
+
           // update ij
           ij += 1;
         } // end j loop
 
-        // For new rescale -EJT, 03-03-2020
-        sum_entprime_flux = -(ith_flux_term1 * eta_prime1 + ith_flux_term2 *
-            eta_prime2 + ith_flux_term3 * eta_prime3);
+        // Finally define it here
+        dij_small = 1E-14 * dij_small;
 
-        // Change rescaling to match TAMU code -EJT, 12/9/2019
+        // Change rescaling to match TAMU code -EJT
         // small_recale=0.5*g*eps*H_{0,max}^2
         double small_rescale = g * hEps * hEps / 1E-5;
         double rescale = fmax(fabs(etaMax[i] - etaMin[i]) / 2., small_rescale);
 
         // new rescale factor = max(|ent_flux_sum| + |-ent'*flux|, 0.0)
-        double new_rescale = fmax(fabs(entropy_flux) + fabs(sum_entprime_flux), 1E-300);
+        sum_entprime_flux =
+            -(ith_flux_term1 * eta_prime1 + ith_flux_term2 * eta_prime2 +
+              ith_flux_term3 * eta_prime3);
+        double new_rescale =
+            fmax(fabs(entropy_flux) + fabs(sum_entprime_flux), 1E-30);
 
         // COMPUTE ENTROPY RESIDUAL //
-        double one_over_entNormFactori = 1.0 / rescale;
+        double one_over_entNormFactori = 1.0 / new_rescale;
```

Comment:
will there be a use for old rescale?

Reply:
I'm not sure at the moment. I left it there just in case I want to use it again. The "new_rescale" is what is implemented in the TAMU code, but still not 100% sure on the benefits. They seem to perform similarly in the "eye" norm.